### PR TITLE
Add .gitattributes file to hide generated files diffs

### LINKED ISSUE #2673: Set up .gitattributes to mark generated files as such
Some of the files are generated by scripts (like `package-lock.json`, the GraphQL schema, `gql-types.ts` and so on), we should tell GitHub that those are generated to make them appear collapsed in pull requests by default. I don't think we need to mark them as binary though.

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
Closes #2673.

This hides diffs from:
- Pipenv lock files;
- GraphQL generated files (types and schemas);
- mjml precompiled files;
- django generated migration files;
- pot files;
- svg files.

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Package manager lock files
+Pipfile.lock linguist-generated
+package-lock.json linguist-generated
+
+# Generated GraphQL schemas
+*.graphql linguist-generated
```
Comment (patrys) [inline .gitattributes]: I'm 👎 on this particular change. That's the only way to understand how a backend change affects the API.

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+# Generated GraphQL types
+gql-types.ts linguist-generated
```
Comment (patrys) [inline .gitattributes]: I think we got rid of that?

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+# Generated django migration files
+*_auto_*.py linguist-generated
+
+# Generated locale pot files
+*.po linguist-generated
```
Comment (patrys) [inline .gitattributes]: These are not really generated.

File: .gitattributes [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Package manager lock files
+Pipfile.lock linguist-generated
+package-lock.json linguist-generated
```
Comment (patrys) [inline .gitattributes]: Could we also add `-diff` and `-merge` for these? It rarely makes sense to resolve these by hand.

File: .gitattributes (hunk 3 above)
Comment (NyanKiyoshi) [reply .gitattributes]: How do we want to handle them? Do we want to show the diffs?

Comment (patrys) [reply .gitattributes]: I'd like to see a diff on GitHub but would explicitly tell git not to attempt a three-way merge.

**Comment by NyanKiyoshi** [comment]
@Pacu2 you can remove `Help Wanted`, thanks again.

**Comment by NyanKiyoshi** [comment]
@patrys I updated the changes, let me know if it's good now or not.

**Pull request merged**